Autotested Results for E2 submitted by ff fl (failures)
Generated at 2015-07-21T12:34:14

Summary of Results: 27 out of 32 tests successfully passed
--

Tests for E2.e2tester.GraphTest (24/29 passed)
--

1) the addEdge(int id1, int id2) method                                               .. ok!
2) the areAdjacent(Node<T> node1, Node<T> node2) method                               .. ok!
3) the areAdjacent(int id1, int id2) method                                           .. ok!
4) the getNode(int id) method                                                         .. ok!
5) the addEdge(int id1, int id2) method                                               .. ok!
6) the getNumEdges() method                                                           .. ok!
7) the getNumNodes() method                                                           .. ok!
8) the addEdge(int id1, int id2) method                                               .. ok!
9) the addEdge(int id1, int id2) method                                               .. ok!
10) the areAdjacent(Node<T> node1, Node<T> node2) method                               .. ok!
11) the areAdjacent(int id1, int id2) method                                           .. ok!
12) the getNodes() method                                                              .. ok!
13) the getNeighbours(Node<T> node) method                                             .. ok!
14) a Graph with no Nodes                                                              .. ok!
15) the addEdge(Node<T> node1, Node<T> node2) method                                   .. ok!
16) the Graph constructor                                                              .. ok!
17) the getNode(int id) method                                                         .. ok!
18) the addNode(int id, T value) method                                                .. ok!
19) the areAdjacent(int id1, int id2) method                                           .. ok!
20) the toString() method                                                              .. ok!
21) a Graph with multiple Nodes                                                        .. ok!
22) the areAdjacent(Node<T> node1, Node<T> node2) method                               .. ok!
23) the areAdjacent(int id1, int id2) method                                           .. ok!
24) the areAdjacent(Node<T> node1, Node<T> node2) method                               .. ok!
25) a Graph with several Nodes                                                         .. failed
.. because Two Nodes should be adjacent when there is an edge between them (details below):

    java.lang.AssertionError: Two Nodes should be adjacent when there is an edge between them
    	at org.junit.Assert.fail(Assert.java:88)
    	at org.junit.Assert.assertTrue(Assert.java:41)
    	at E2.e2tester.GraphTest.testGraphMultipleNodesByID(GraphTest.java:290)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:606)
    	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    	at java.lang.Thread.run(Thread.java:745)

26) the addEdge(Node<T> node1, Node<T> node2) method                                   .. failed
.. because java.lang.NullPointerException (details below):

    java.lang.NullPointerException
    	at e2soln.Graph.addEdge(Graph.java:141)
    	at E2.e2tester.GraphTest.testAddEdgeNode1NotInGraph(GraphTest.java:563)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:606)
    	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    	at java.lang.Thread.run(Thread.java:745)

27) the addEdge(Node<T> node1, Node<T> node2) method                                   .. failed
.. because java.lang.NullPointerException (details below):

    java.lang.NullPointerException
    	at e2soln.Graph.addEdge(Graph.java:142)
    	at E2.e2tester.GraphTest.testAddEdgeNode2NotInGraph(GraphTest.java:582)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:606)
    	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    	at java.lang.Thread.run(Thread.java:745)

28) a Graph with a single Node                                                         .. failed
.. because no self edges are allowed (details below):

    java.lang.AssertionError: no self edges are allowed
    	at org.junit.Assert.fail(Assert.java:88)
    	at org.junit.Assert.assertTrue(Assert.java:41)
    	at E2.e2tester.GraphTest.testGraphOneNode(GraphTest.java:212)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:606)
    	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    	at java.lang.Thread.run(Thread.java:745)

29) the addEdge(Node<T> node1, Node<T> node2) method                                   .. failed
.. because java.lang.NullPointerException (details below):

    java.lang.NullPointerException
    	at e2soln.Graph.addEdge(Graph.java:141)
    	at E2.e2tester.GraphTest.testAddEdgeNodeNoNodesInGraph(GraphTest.java:598)
    	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    	at java.lang.reflect.Method.invoke(Method.java:606)
    	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
    	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    	at java.lang.Thread.run(Thread.java:745)



Tests for E2.e2tester.NoSuchNodeExceptionTest (3/3 passed)
--

1) that NoSuchNodeException is a checked exception                                    .. ok!
2) the NoSuchNodeException constructor with one argument                              .. ok!
3) the NoSuchNodeException constructor with no arguments                              .. ok!


